src/components/views/sidebar/TreeView/InstanceTreeView.js

```diff
@@ -0,0 +1,384 @@
+    if (Array.isArray(treeItems) && treeItems.length <= 0) return;
+
+    return treeItems?.map((treeItemData) => {
+      let items = undefined;
+      if (treeItemData.items && treeItemData.items.length > 0) {
+        items = getTreeItemsFromData(treeItemData.items);
+      }
+
+      const hidden =
+        !nodes.includes(treeItemData?.id) &&
+        selectedNodeId !== treeItemData?.id &&
+        treeItemData?.type === 'COMPOSITION';
+
+      const labelIcon =
+        treeItemData?.type === 'MECHANISM' ? (
+          <CircleIcon />
+        ) : treeItemData?.type === 'COMPOSITION' ? (
+          <FileIcon />
+        ) : (
+          <ShapeArrowToolIcon />
+        );
+
+      return (
+        <StyledTreeItem
+          dataset={treeItemData?.id}
+          nodeId={treeItemData?.id}
+          labelText={treeItemData?.label}
+          labelIcon={labelIcon}
+          children={items}
+          key={treeItemData?.id}
+          selected={treeItemData?.id === selectedNodeId}
+          onNodeSelect={onNodeSelect}
+          onShowPanel={onRightClick}
+          hidden={hidden}
+          type={treeItemData?.type}
+        />
+      );
+    });
+  };
+
+  const functionValues = (label, value) => (
+    <Box className={[classes.block, classes.paddingXS]}>
+      <Typography component="label">{label}</Typography>
+      <Typography className="function" noWrap>
+        {value}
+      </Typography>
+    </Box>
+  );
+
+  const treeRef = React.createRef();
+
+  const id = open ? 'simple-popper' : undefined;
+
+  return (
+    <>
+      <TreeView
+        className="scrollbar treeViewer"
+        defaultExpanded={nodes}
+        defaultCollapseIcon={
+          <ArrowDropDownIcon fontSize="small" color="disabled" />
+        }
+        defaultExpandIcon={<ArrowRightIcon fontSize="small" color="disabled" />}
+        defaultEndIcon={false}
+        ref={treeRef}
+        expanded={nodes}
+        onNodeToggle={onNodeToggle}
+        selected={[selectedNodeId]}
+        onNodeSelect={(e, node_ids) => {
+          handleClick(e, node_ids);
+        }}
+      >
+        {getTreeItemsFromData(items)}
+      </TreeView>
+      <Popover
+        id={id}
+        open={open}
+        anchorReference="anchorPosition"
+        anchorPosition={
+          right.mouseY !== null && right.mouseX !== null
+            ? { top: right.mouseY, left: right.mouseX }
+            : undefined
+        }
+        onClose={onClose}
+        componentsProps={{
+          backdrop: {
+            invisible: true,
+            onContextMenu: (event) => {
+              event.preventDefault();
+              event.stopPropagation();
+
+              // close popover
+              onClose();
+            },
+          },
+        }}
+        PaperProps={popperPaperProps}
+      >
+        <Box className={classes.popperHeader}>
+          <Box>
+            <FileIcon color="black" />
+            <Typography component="strong" noWrap>
+              Composition 2
```

Comment:
dont hardcode

Reply:
The core task is focusing on UI as there no insight of what the data should be yet and will make modifications regardless

Reply:
@emekauja please try to avoid this and use constants / constant lists and iteration over the lists because it will make the implementation on a later stage easier

Reply:
we avoid this totally going forward thank you Zoran